pkg/ottl/ottlfuncs/func_contains.go

```diff
@@ -0,0 +1,46 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package ottlfuncs // import "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl/ottlfuncs"
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl"
+)
+
+type ContainsArguments[K any] struct {
+	Target []ottl.StringLikeGetter[K]
+	Item   string
```

Comment:
I want to ask to check if current approach is fine. There is a limitation that `item` can only be of `string` type, in the current approach I cannot accept the element as `int` or any other type.

I create the target as a slice of a string. If the item should also be of another type, like `int`, `float`, I can work on improving it.  I would appreciate tips on how to do this, I found a `listGetter`, so maybe it will be useful.